src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -215,6 +215,16 @@ private void MethodChecks(MethodDeclarationSyntax syntax, Binder withTypeParamsB
                 diagnostics.Add(ErrorCode.WRN_FinalizeMethod, location);
             }
 
+            // Any nullable typeParameter declared by the method in the signature of an override or explicit interface implementation is considered a Nullable<T>
+            if (IsExplicitInterfaceImplementation || IsOverride)
```

Comment:
>IsExplicitInterfaceImplementation [](start = 16, length = 33)

```syntax.ExplicitInterfaceSpecifier != null```?

Reply:
I wasn't sure which to use. What is the difference?

Reply:
> I wasn't sure which to use. What is the difference?

It is good to be consistent in a way this functions checks for certain conditions. For example, below it is using syntax to check if this is an explicit interface implementation. This check will also be faster, eventually the same syntax condition will be checked, I think, but it will jump through more hoops. Etc.

---
In reply to: [275009353](https://github.com/dotnet/roslyn/pull/34584#discussion_r275009353) [](ancestors = 275009353)